tests/docs/smoke-all/2023/05/31/issue-5506.qmd

```diff
@@ -0,0 +1,38 @@
+---
+title: plots and output = "asis"
+_quarto:
+  tests:
+    markdown:
+      ensureFileRegexMatches:
+        - [] 
+        - ["::: cell-output-display"]
+        - ["* one\n* two"]
+        - ["* three\n* four"]
```

Comment:
The syntax is 
````yaml
ensureFileRegexMatches:    
    - ["regex_to_match", "another_regex to match"]
    - ["regex_to_not_match", "another regex_to_not_match"]
````

So `ensureFileRegexMatches` takes 
* First a list of regex that should be match - if they are not match, then the test will error
* Second a list of regex that should not be match - if they are the test will error

It is indeed not documented so you couldn't know without reading the `.ts` file. 

Some tips for future PR: Using `html` as a format allows you to use `ensureHtmlElements` which works similarly by using CSS selectors - this can sometime be easier than regex.

Reply:
Thanks for the information!

I tried to have a go at this but am not at all familiar with the unit testing notation. When I render the test qmd manually it looks good.

Would you be willing to take a look at what I'm doing wrong with the unit test?

Reply:
Sure. I'll make time later to look at it. I am at a conference this week, so a bit slower than usual, but I'll have a look.  

I think you just need to escape properly the regex and / or use some CSS selector instead of regex.

Reply:
Thanks for the tip; I managed to solve the issue. Since I managed to solve fix the unit test, I'll consider this issue resolved and will request a re-review :)

Enjoy your conference!